src/picterra/__main__.py

```diff
@@ -134,9 +150,31 @@ def parse_args(args):
         client = APIClient()
         logger.info('Running %s on %s' % (options.detector, options.raster))
         logger.debug('Starting detection..')
-        result_id = client.run_detector(options.detector, options.raster)
-        client.download_result_to_file(result_id, options.output_file)
-        logger.debug('Detection finished, writing result to %s' % options.output_file)
+        operation_id = client.run_detector(options.detector, options.raster)
+        if options.output_type == 'geometries':   # outputting/saving result geometries
+            if options.output_file:
+                logger.debug('Detection finished, writing result to %s' % options.output_file)
+                client.download_result_to_file(operation_id, options.output_file)
+            else:
+                from tempfile import mkstemp
+                fd, path = mkstemp()
+                client.download_result_to_file(operation_id, path)
+                logger.debug('Detection finished, outputting result data')
+                with open(path) as f:
+                    for piece in _read_in_chunks(f):
+                        # return values
+                        print(piece, end='')
+                # close the file descriptor
+                os.close(fd)
```

Comment:
Is this enough? I just checked `mkstemp`'s [doc](https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp) and it says: `Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.` Should we `unlink` it as well?

Reply:
Checked